plugins/woocommerce/src/Admin/API/Reports/Coupons/DataStore.php

```diff
@@ -64,8 +64,10 @@ protected function assign_report_columns() {
 
 	/**
 	 * Set up all the hooks for maintaining and populating table data.
+	 *
+	 * @internal
 	 */
-	public static function init() {
+	final public static function init() {
```

Comment:
Is this needed? could be a backward incompatible change.

Reply:
All the `init` methods in classes in `src` need to be final, it's a requirement enforced by the linter (we introduced that requirement together with the dependency injection container).

Reply:
I see, but if we have already released it then it would be backward incompatible if someone is overwriting this class.

Reply:
Ok, I have removed it and added a linter exception.